### PR TITLE
Update Blog Layout

### PR DESCRIPTION
**Pull request by hramos**
Updated the blog's styling to make it more readable.

* BlogPageLayout (blog index) - Only excerpts from each article are now shown, as opposed to the entire article.
* BlogPost - Broken up into headers, footers. Reorder header so that the blog post title is closer to the content. Adds support for hero images (visible from the blog index). Adds Facebook, Twitter social share buttons. List items are properly spaced now. 

Blog index:
screencapture-localhost-8079-react-native-blog-1471905976431

Single blog post:
screencapture-localhost-8079-react-native-blog-2016-08-12-react-native-meetup-san-francisco-html-1471905997923

### CODE REVIEW

**Comment by facebook-github-bot-5** [comment]
By analyzing the blame information on this pull request, we identified @Andreyco and @JoelMarcey to be potential reviewers.

**Comment by hramos** [comment]
cc @bestander, @mkonicek who may want to take a look.

**Comment by JoelMarcey** [comment]
I took a look and think this can be shipped. But I will wait for others to see if they have any opinion.

**Comment by mkonicek** [comment]
Looks awesome!

**Comment by bestander** [comment]
Circle CI test can be ignored

File: website/core/Site.js [hunk 1]
```diff
@@ -46,7 +46,8 @@ var Site = React.createClass({
           <script type="text/javascript">{'try{Typekit.load();}catch(e){}'}</script>
         </head>
         <body>
-
+          <script dangerouslySetInnerHTML={{__html: `window.fbAsyncInit = function() {FB.init({appId:'1677033832619985',xfbml:true,version:'v2.7'});};(function(d, s, id){var js, fjs = d.getElementsByTagName(s)[0];if (d.getElementById(id)) {return;}js = d.createElement(s); js.id = id;js.src = '//connect.facebook.net/en_US/sdk.js';fjs.parentNode.insertBefore(js, fjs);}(document, 'script','facebook-jssdk'));`}} />
```
Comment (bestander) [inline website/core/Site.js]: I hope you know what you are doing :)

**Comment by bestander** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-0** [comment]
Thanks for importing.If you are an FB employee go to [Phabricator](https://phabricator.intern.facebook.com/D3758524) to review internal test results.

**Pull request closed without merging**